games-emulation/rmg: don't hardcode pkg-config + fix -Wstrict-aliasing warnings

orbea opened the pull request:
* Only call rust_pkg_setup when using rust
Fixes: https://github.com/gentoo/gentoo/commit/01ba4dc61965ef7658a24728841c04c9a1ce4871
* Use `pkg-config` for SDL2 in mupen64plus-core (Don't use `which`)
Upstream-PR: https://github.com/mupen64plus/mupen64plus-core/pull/1093
Upstream-Commit: https://github.com/mupen64plus/mupen64plus-core/commit/966a5df3165afede21204610c396dba1ed8a5bf3
Upstream-Commit: https://github.com/mupen64plus/mupen64plus-core/commit/b007759b9ed9d1473c732b4f3abcd9257165c65c
Upstream-Commit: https://github.com/Rosalie241/RMG/commit/4cd4f4c6ea37c82cdda6b0a57e79af1b71df17fe
* Fixes hard-coded `pkg-config`.
Closes: https://bugs.gentoo.org/941889
Upstream-PR: https://github.com/raphnet/mupen64plus-input-raphnetraw/pull/17
Upstream-PR: https://github.com/Rosalie241/RMG/pull/296
Upstream-PR: https://github.com/Rosalie241/RMG/pull/2970
Upstream-Commit: https://github.com/Rosalie241/RMG/commit/f6b73b86c6625fa5a268b74b1e36114a5bdd535c
Upstream-Commit: https://github.com/Rosalie241/RMG/commit/69c24b9131909c338b03c48438744b253501146a
* Fix `-Wstrict-aliasing` warnings
Upstream-Issue: https://github.com/gonetz/GLideN64/issues/2877
Upstream-PR: https://github.com/gonetz/GLideN64/pull/2884
Upstream-Commit: https://github.com/gonetz/GLideN64/commit/a1320da566c03503ea755447458bb2b79a610cb5
Upstream-Commit: https://github.com/Rosalie241/RMG/commit/e68483fce2a977c07fee38e1d8d8f8dc22bb6072

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

eli-schwartz commented on games-emulation/rmg/files/rmg-0.6.6-mupen64plus-input-raphnetraw-pkgconfig.patch:
```diff
@@ -0,0 +1,46 @@
++# test for essential build dependencies
++ifeq ($(origin PKG_CONFIG), undefined)
++  PKG_CONFIG = $(CROSS_COMPILE)pkg-config
++  ifeq ($(shell which $(PKG_CONFIG) 2>/dev/null),)
```
I reviewed this in the (merged) upstream PR. It is not a great solution -- in order to use this patch, you'd have to add a new BDEPEND="sys-apps/which"

orbea replied:
I was copying the check from elsewhere in the project, but I believe because `PKG_CONFIG` is set in the ebuild this code will never be reached by Gentoo.

orbea replied:
While this is true for the above code I neglected the part right below it so I revised the patch upstream to avoid `which(1)` in mupen64plus-input-raphnetraw. However there appear to be other uses of `which(1)` in mupen64plus-core which I didn't notice before so I added the `sys-apps/which` dependency regardless. I will see about trying to remove it upstream later.

eli-schwartz replied:
Ok thanks. I don't love depending on it at all but it's hardly the only package to do it. The most important thing at the moment is actually just to get a good model of what packages actually have to have it installed as part of https://bugs.gentoo.org/show_bug.cgi?id=which-hunt

orbea replied:
I agree that it would be better to not depend on it and made this PR for mupen64plus-core that would avoid it in Gentoo, but I will have to see what they think.

https://github.com/mupen64plus/mupen64plus-core/pull/1093

orbea replied:
I updated this PR so that RMG no longer depends on `which` when `PKG_CONFIG` is explicitly set by the ebuild.

I also added the missing dependency to `0.6.5`.

orbea commented:
Rebased the PR after the changes with rust.

orbea commented:
Also added a commit to fix a change in commit https://github.com/gentoo/gentoo/commit/01ba4dc61965ef7658a24728841c04c9a1ce4871.

eli-schwartz commented:
> Also added a commit to fix a change in commit [01ba4dc](https://github.com/gentoo/gentoo/commit/01ba4dc61965ef7658a24728841c04c9a1ce4871).

That commit doesn't include a signed-off-by, so pram won't even allow me to download it and the tooling hooks would reject it. Can you add a signoff to that commit too?

Also, for `Fixes:` lines it's fairly common to just use the commit sha1 (github still makes this clickable, and it's only a mirror of gitweb anyway).

orbea commented:
> > Also added a commit to fix a change in commit [01ba4dc](https://github.com/gentoo/gentoo/commit/01ba4dc61965ef7658a24728841c04c9a1ce4871).
> 
> That commit doesn't include a signed-off-by, so pram won't even allow me to download it and the tooling hooks would reject it. Can you add a signoff to that commit too?
> 
> Also, for `Fixes:` lines it's fairly common to just use the commit sha1 (github still makes this clickable, and it's only a mirror of gitweb anyway).

Fixed on both counts, I wonder why the CI didn't fail without the sign-off? I believe I have seen it fail in the past?

eli-schwartz commented:
Sorry for the late reply and letting this drop off my radar. :( This is now tested and queued and I will be pushing it momentarily.
